core/src/main/java/de/jplag/reporting/jsonfactory/ComparisonReportWriter.java

```diff
@@ -103,16 +102,9 @@ private Match convertMatchToReportMatch(JPlagComparison comparison, de.jplag.Mat
         Token startOfSecond = tokensSecond.stream().min(lineComparator).orElseThrow();
         Token endOfSecond = tokensSecond.stream().max(lineComparator).orElseThrow();
 
-        return new Match(relativizedFilePath(startOfFirst.getFile(), comparison.firstSubmission()),
-                relativizedFilePath(startOfSecond.getFile(), comparison.secondSubmission()), startOfFirst.getLine(), endOfFirst.getLine(),
-                startOfSecond.getLine(), endOfSecond.getLine(), match.length());
-    }
-
-    private String relativizedFilePath(File file, Submission submission) {
-        if (file.toPath().equals(submission.getRoot().toPath())) {
-            return Path.of(submissionToIdFunction.apply(submission), submissionToIdFunction.apply(submission)).toString();
-        }
-        return Path.of(submissionToIdFunction.apply(submission), submission.getRoot().toPath().relativize(file.toPath()).toString()).toString();
+        return new Match(FilePathUtil.getRelativeSubmissionPath(startOfFirst.getFile(), comparison.firstSubmission(), submissionToIdFunction),
```

Comment:
These three lines really aren't that readable, and neither were the original ones.
That's the only thing I found in my static code review, I will perform some runtime tests later.